src/Microsoft.Framework.OptionsModel/ConfigurationBinder.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Framework.ConfigurationModel;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.Framework.OptionsModel
+{
+    public class ConfigurationBinder
```

Comment:
This whole class pattern is a bit unusual... almost all the methods are static, except for the ctor and one method, and the only field is set once in the ctor and used once in that one method. There don't seem to be any cases where the instance `Bind` method is called more than once. The runtime doesn't seem to do it, the unit tests don't do it. Is that a scenario?